controller/k8s/api.go

```diff
@@ -38,13 +39,35 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
-// APIResource is an enum for Kubernetes API resource types, for use when
+// APIResource represents a namespaces k8s api resource
```

Comment:
I think I would rather that service mirror build it's own secret informer instead of trying to make the existing api.go machinery flexible enough to support this use case.